library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -655,7 +707,18 @@ public void emitTuples()
         } else if (!failedFiles.isEmpty()) {
           retryFailedFile(failedFiles.poll());
         } else {
-          scanDirectory();
+          int fileCt = 0;
+          while ((!fileQueue.isEmpty()) && (fileCt < MAX_FILE_PER_WINDOW)) {
```

Comment:
As mentioned above, I'll make it configurable. The purpose of it is to limit the number of files in the queue when we have large number of files.